src/components/application_manager/src/request_controller.cc

```diff
@@ -364,9 +369,27 @@ void RequestController::onTimer() {
   SDL_AUTO_TRACE();
   SDL_DEBUG(
       "ENTER Waiting fore response count: " << waiting_for_response_.Size());
-  RequestInfoPtr probably_expired =
-      waiting_for_response_.FrontWithNotNullTimeout();
-  while (probably_expired && probably_expired->isExpired()) {
+  while(!stop_flag_) {
```

Comment:
@LuxoftAKutsan Sorry for delay reply, I missed the mail. New sequence diagram please reference [**here**](https://www.websequencediagrams.com/cgi-bin/cdraw?lz=dGl0bGUgUmVxdWVzdCBDb250cm9sbGVyCgpBTS0-ABEHQ3RybDogQWRkACEICgAOCwAXD3B1dCByAEgHaW4gcXVldWUAIQ5UaW1lcjogU3RhcnQgAAgFCgpsb29wIG5vdCB0aW1lcl9zdG9wX2ZsYWcKCm9wdAA_BiBpcyBlbXB0eQoAPAUAPwlBdXRvIGxvY2sgd2FpdAplbmQALwZub3QgaXNFeHBpcmVkACUPd2FpdCBmb3IgbXNlY3MALwUAMwUAgRQUVXBkYXRlAIETBgBsD1dhaXQAgXE4AIF_KQBgIACCBwcAgx4Nb25UaW1lb3V0AIFEDwCDQw1Qcm9jZXNzIGZpcnMAAYELAHkTIChsYXN0IG9uZSkAhDUWdXNwZW5kAIMPFQCEEA8AhSAaRXhpAIVJDwCFIgcAhQQPID0gdHJ1ZQoK&s=rose), I think the most logic is not changed.